Applications/Spire/Include/Spire/Ui/ColorConversion.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef SPIRE_COLOR_CONVERSION_HPP
+#define SPIRE_COLOR_CONVERSION_HPP
+#include <QColor>
+
+namespace Spire {
+
+  /** Represents a Oklab color. */
+  struct OklabColor {
+
+    /** Specifies if the color is valid. */
+    bool m_is_valid;
```

Comment:
When would this be false?

Reply:
When `QColor` converts to `OklabColor` and `QColor` is invalid, `OklabColor` will be invalid and `m_is_valid` will be false.

Reply:
Okay, in that case write a function to check if an ```OklabColor``` is valid, and reserve some kind of representation for an invalid ```OklabColor```, for example if any component is ```-1``` then ```is_valid``` returns ```false```.